pandora-server-directory/src/account/accountDirectMessages.ts

```diff
@@ -69,6 +85,27 @@ export class AccountDirectMessages {
 				delete dm.closed;
 				break;
 		}
+		if (this._infos) {
```

Comment:
This can lead to a race condition with `getDirectMessageInfo`. Both should use an `object` async synchronization.